### ISSUE TITLE
Invalid token in combination with ajax operations should result in warnings.

### ISSUE THREAD

**Issue by musicpanda** [PrestaShop/PrestaShop#38821]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When you do an ajax operation (for example uninstalling a module) at a time when the token is invalid you get a message that the operation failed. You are not warned that the cause was that an invalid token.

This is extremely confusing and regularly results in people overreacting and causing damage elsewhere in the shop. 

### Expected behavior

A popup should appear that reports the token problem.

### Steps to reproduce

Disable a module when the token has expired.

### PrestaShop version(s) where the bug happened

8.2

**Comment by SiraDIOP** [comment]
Hello @musicpanda ,

Thanks for reporting this issue!
Ping @PrestaShop/committers : Could someone please try to reproduce the issue, it's too technical!

Thanks in advance!

**Comment by boo-code** [comment]
The cause is visible in `TokenizedUrlsListener::onKernelRequest()` on `9.2.x`: an invalid token always produces a redirect, with no branch for ajax.

```php
if (!$this->userTokenManager->isTokenValid()) {
    $uri = UrlCleaner::cleanUrl($request->getRequestUri(), ['_token', 'token']);
    $uri = $request->getSchemeAndHttpHost() . $uri;
    $response = new RedirectResponse($this->router->generate('admin_security_compromised', ['uri' => urlencode($uri)]));
    $event->setResponse($response);
}
```

For a normal page that is the right behaviour and the merchant lands on the warning page, which explains itself. For an ajax call the browser follows the redirect and the script receives the HTML of that page where it expected JSON, so it reports only that the operation failed. Nothing in that path ever says "token", which matches your description and explains why the reaction is to go looking for other causes.

The shape of the fix is to answer ajax differently, for instance:

```php
if ($request->isXmlHttpRequest()) {
    $event->setResponse(new JsonResponse(
        ['errors' => [/* invalid token message */]],
        Response::HTTP_FORBIDDEN
    ));

    return;
}
```

One thing I could not check, and it decides whether this is enough on its own: whether the calling script surfaces `errors` from a failed response, or shows its own generic message regardless. If it is the latter then the back office javascript needs a change too, and fixing only the listener would leave the merchant with the same unhelpful text.

So the back end part is small and clear, and someone who can open the module manager should confirm what the script does with a 403 carrying `errors` before it is called done. I am happy to write the listener change if a maintainer confirms that side.
